src/status_im/components/icons/vector_icons.cljs

```diff
@@ -24,66 +24,71 @@
             :icons/exclamation_mark    (slurp-svg "./resources/icons/exclamation_mark.svg")
             :icons/forward             (slurp-svg "./resources/icons/forward.svg")
             :icons/fullscreen          (slurp-svg "./resources/icons/fullscreen.svg")
-            :icons/group_big           (slurp-svg "./resources/icons/group_big.svg")
-            :icons/group_chat          (slurp-svg "./resources/icons/group_chat.svg")
+            :icons/group-big           (slurp-svg "./resources/icons/group_big.svg")
+            :icons/group-chat          (slurp-svg "./resources/icons/group_chat.svg")
             :icons/hamburger           (slurp-svg "./resources/icons/hamburger.svg")
             :icons/hidden              (slurp-svg "./resources/icons/hidden.svg")
             :icons/mic                 (slurp-svg "./resources/icons/mic.svg")
             :icons/ok                  (slurp-svg "./resources/icons/ok.svg")
             :icons/public              (slurp-svg "./resources/icons/public.svg")
-            :icons/public_chat         (slurp-svg "./resources/icons/public_chat.svg")
+            :icons/public-chat         (slurp-svg "./resources/icons/public_chat.svg")
             :icons/qr                  (slurp-svg "./resources/icons/QR.svg")
             :icons/search              (slurp-svg "./resources/icons/search.svg")
             :icons/smile               (slurp-svg "./resources/icons/smile.svg")
-            :icons/commands_list       (slurp-svg "./resources/icons/commands_list.svg")
-            :icons/dropdown_up         (slurp-svg "./resources/icons/dropdown_up.svg")
+            :icons/commands-list       (slurp-svg "./resources/icons/commands_list.svg")
+            :icons/dropdown-up         (slurp-svg "./resources/icons/dropdown_up.svg")
             :icons/dropdown            (slurp-svg "./resources/icons/dropdown.svg")
             :icons/grab                (slurp-svg "./resources/icons/grab.svg")})
 
-(defn check-name [n]
+(defn normalize-property-name [n]
   (if (= n :icons/options)
-    (if ios? :icons/dots_horizontal :icons/dots_vertical)
+    (if ios? :icons/dots-horizontal :icons/dots-vertical)
     n))
 
-(defn icon [name & [opts]]
-  (let [{:keys [color container-style style]} opts]
-    [rn/view container-style
-     (into []
-       (concat
-         [svg (merge {:width 24 :height 24 :viewBox "0 0 24 24"} style)]
-         ((get icons (check-name name))
-          (case color
-            :dark common/icon-dark-color
-            :gray common/icon-gray-color
-            :blue common/color-light-blue
-            :active common/color-blue4
-            :white common/color-white
-            :red common/icon-red-color
-            common/icon-dark-color))))]))
+(def default-viewbox {:width 24 :height 24 :viewBox "0 0 24 24"})
+
+(defn icon
+  ([name] (icon name nil))
+  ([name {:keys [color container-style style]}]
+   [rn/view container-style
+    (if-let [icon (get icons (normalize-property-name name))]
```

Comment:
Is the shadowing of icon on purpose? Can we disambiguate them? Also not 100% clear on how this works, with `icon` fn calling `icon` recursively. Could you clarify code or perhaps write a short comment?

Reply:
Right the shadowinf is unfortunate. Nested `icon` is actually a function.